File: frontend/src/types/InterventionPlan.ts
export interface InterventionPlan {
  planId: number;
  residentId: number;
  planCategory?: string;
  planDescription?: string;
  servicesProvided?: string;
  targetValue?: number;
  targetDate?: string;
  status?: string;
  caseConferenceDate?: string;
  createdAt?: string;
  updatedAt?: string;
}

File: backend/Beacon.API/Models/InterventionPlan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Beacon.API.Models;

public partial class InterventionPlan
{
    [Key]
    public int PlanId { get; set; }

    public int ResidentId { get; set; }

    public string? PlanCategory { get; set; }

    public string? PlanDescription { get; set; }

    public string? ServicesProvided { get; set; }

    public decimal? TargetValue { get; set; }

    public DateOnly? TargetDate { get; set; }

    public string? Status { get; set; }

    public DateOnly? CaseConferenceDate { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual Resident Resident { get; set; } = null!;
}
